File: qt/multimedia/gen_qmediacapturesession.go
package multimedia

import (
	"unsafe"

	"github.com/mappu/miqt/qt"
)

type QMediaCaptureSession struct {
	h          uintptr
	isSubclass bool
}

// NewQMediaCaptureSession constructs a new QMediaCaptureSession object.
func NewQMediaCaptureSession() *QMediaCaptureSession {
	ret := newQMediaCaptureSession(QMediaCaptureSession_new())
	ret.isSubclass = true
	return ret
}

// NewQMediaCaptureSession2 constructs a new QMediaCaptureSession object.
func NewQMediaCaptureSession2(parent *qt.QObject) *QMediaCaptureSession {
	ret := newQMediaCaptureSession(QMediaCaptureSession_new2((*QObject)(parent.UnsafePointer())))
	ret.isSubclass = true
	return ret
}

func (this *QMediaCaptureSession) MetaObject() *qt.QMetaObject {
	return qt.UnsafeNewQMetaObject(unsafe.Pointer(QMediaCaptureSession_MetaObject(this.h)))
}

func (this *QMediaCaptureSession) Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))
	return (unsafe.Pointer)(QMediaCaptureSession_Metacast(this.h, param1_Cstring))
}

func QMediaCaptureSession_Tr(s string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	var _ms struct_miqt_string = QMediaCaptureSession_Tr(s_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QMediaCaptureSession) AudioInput() *QAudioInput {
	return newQAudioInput(QMediaCaptureSession_AudioInput(this.h))
}

func (this *QMediaCaptureSession) SetAudioInput(input *QAudioInput) {
	QMediaCaptureSession_SetAudioInput(this.h, input.cPointer())
}

func (this *QMediaCaptureSession) AudioBufferInput() *QAudioBufferInput {
	return newQAudioBufferInput(QMediaCaptureSession_AudioBufferInput(this.h))
}

func (this *QMediaCaptureSession) SetAudioBufferInput(input *QAudioBufferInput) {
	QMediaCaptureSession_SetAudioBufferInput(this.h, input.cPointer())
}

func (this *QMediaCaptureSession) Camera() *QCamera {
	return newQCamera(QMediaCaptureSession_Camera(this.h))
}

func (this *QMediaCaptureSession) SetCamera(camera *QCamera) {
	QMediaCaptureSession_SetCamera(this.h, camera.cPointer())
}

func (this *QMediaCaptureSession) ImageCapture() *QImageCapture {
	return newQImageCapture(QMediaCaptureSession_ImageCapture(this.h))
}

func (this *QMediaCaptureSession) SetImageCapture(imageCapture *QImageCapture) {
	QMediaCaptureSession_SetImageCapture(this.h, imageCapture.cPointer())
}

func (this *QMediaCaptureSession) ScreenCapture() *QScreenCapture {
	return newQScreenCapture(QMediaCaptureSession_ScreenCapture(this.h))
}

func (this *QMediaCaptureSession) SetScreenCapture(screenCapture *QScreenCapture) {
	QMediaCaptureSession_SetScreenCapture(this.h, screenCapture.cPointer())
}

func (this *QMediaCaptureSession) WindowCapture() *QWindowCapture {
	return newQWindowCapture(QMediaCaptureSession_WindowCapture(this.h))
}

func (this *QMediaCaptureSession) SetWindowCapture(windowCapture *QWindowCapture) {
	QMediaCaptureSession_SetWindowCapture(this.h, windowCapture.cPointer())
}

func (this *QMediaCaptureSession) VideoFrameInput() *QVideoFrameInput {
	return newQVideoFrameInput(QMediaCaptureSession_VideoFrameInput(this.h))
}

func (this *QMediaCaptureSession) SetVideoFrameInput(input *QVideoFrameInput) {
	QMediaCaptureSession_SetVideoFrameInput(this.h, input.cPointer())
}

func (this *QMediaCaptureSession) Recorder() *QMediaRecorder {
	return newQMediaRecorder(QMediaCaptureSession_Recorder(this.h))
}

func (this *QMediaCaptureSession) SetRecorder(recorder *QMediaRecorder) {
	QMediaCaptureSession_SetRecorder(this.h, recorder.cPointer())
}

func (this *QMediaCaptureSession) SetVideoOutput(output *qt.QObject) {
	QMediaCaptureSession_SetVideoOutput(this.h, (*QObject)(output.UnsafePointer()))
}

func (this *QMediaCaptureSession) VideoOutput() *qt.QObject {
	return qt.UnsafeNewQObject(unsafe.Pointer(QMediaCaptureSession_VideoOutput(this.h)))
}

func (this *QMediaCaptureSession) SetVideoSink(sink *QVideoSink) {
	QMediaCaptureSession_SetVideoSink(this.h, sink.cPointer())
}

func (this *QMediaCaptureSession) VideoSink() *QVideoSink {
	return newQVideoSink(QMediaCaptureSession_VideoSink(this.h))
}

func (this *QMediaCaptureSession) SetAudioOutput(output *QAudioOutput) {
	QMediaCaptureSession_SetAudioOutput(this.h, output.cPointer())
}

func (this *QMediaCaptureSession) AudioOutput() *QAudioOutput {
	return newQAudioOutput(QMediaCaptureSession_AudioOutput(this.h))
}

func (this *QMediaCaptureSession) AudioInputChanged() {
	QMediaCaptureSession_AudioInputChanged(this.h)
}

func (this *QMediaCaptureSession) OnAudioInputChanged(slot func()) {
	QMediaCaptureSession_connect_AudioInputChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_AudioInputChanged
func miqt_exec_callback_QMediaCaptureSession_AudioInputChanged(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QMediaCaptureSession) AudioBufferInputChanged() {
	QMediaCaptureSession_AudioBufferInputChanged(this.h)
}

func (this *QMediaCaptureSession) OnAudioBufferInputChanged(slot func()) {
	QMediaCaptureSession_connect_AudioBufferInputChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_AudioBufferInputChanged
func miqt_exec_callback_QMediaCaptureSession_AudioBufferInputChanged(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QMediaCaptureSession) CameraChanged() {
	QMediaCaptureSession_CameraChanged(this.h)
}

func (this *QMediaCaptureSession) OnCameraChanged(slot func()) {
	QMediaCaptureSession_connect_CameraChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_CameraChanged
func miqt_exec_callback_QMediaCaptureSession_CameraChanged(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QMediaCaptureSession) ScreenCaptureChanged() {
	QMediaCaptureSession_ScreenCaptureChanged(this.h)
}

func (this *QMediaCaptureSession) OnScreenCaptureChanged(slot func()) {
	QMediaCaptureSession_connect_ScreenCaptureChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_ScreenCaptureChanged
func miqt_exec_callback_QMediaCaptureSession_ScreenCaptureChanged(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QMediaCaptureSession) WindowCaptureChanged() {
	QMediaCaptureSession_WindowCaptureChanged(this.h)
}

func (this *QMediaCaptureSession) OnWindowCaptureChanged(slot func()) {
	QMediaCaptureSession_connect_WindowCaptureChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_WindowCaptureChanged
func miqt_exec_callback_QMediaCaptureSession_WindowCaptureChanged(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QMediaCaptureSession) VideoFrameInputChanged() {
	QMediaCaptureSession_VideoFrameInputChanged(this.h)
}

func (this *QMediaCaptureSession) OnVideoFrameInputChanged(slot func()) {
	QMediaCaptureSession_connect_VideoFrameInputChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_VideoFrameInputChanged
func miqt_exec_callback_QMediaCaptureSession_VideoFrameInputChanged(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QMediaCaptureSession) ImageCaptureChanged() {
	QMediaCaptureSession_ImageCaptureChanged(this.h)
}

func (this *QMediaCaptureSession) OnImageCaptureChanged(slot func()) {
	QMediaCaptureSession_connect_ImageCaptureChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_ImageCaptureChanged
func miqt_exec_callback_QMediaCaptureSession_ImageCaptureChanged(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QMediaCaptureSession) RecorderChanged() {
	QMediaCaptureSession_RecorderChanged(this.h)
}

func (this *QMediaCaptureSession) OnRecorderChanged(slot func()) {
	QMediaCaptureSession_connect_RecorderChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_RecorderChanged
func miqt_exec_callback_QMediaCaptureSession_RecorderChanged(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QMediaCaptureSession) VideoOutputChanged() {
	QMediaCaptureSession_VideoOutputChanged(this.h)
}

func (this *QMediaCaptureSession) OnVideoOutputChanged(slot func()) {
	QMediaCaptureSession_connect_VideoOutputChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_VideoOutputChanged
func miqt_exec_callback_QMediaCaptureSession_VideoOutputChanged(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func (this *QMediaCaptureSession) AudioOutputChanged() {
	QMediaCaptureSession_AudioOutputChanged(this.h)
}

func (this *QMediaCaptureSession) OnAudioOutputChanged(slot func()) {
	QMediaCaptureSession_connect_AudioOutputChanged(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_AudioOutputChanged
func miqt_exec_callback_QMediaCaptureSession_AudioOutputChanged(cb intptr_t) {
	gofunc, ok := cgo.Handle(cb).Value().(func())
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	gofunc()
}

func QMediaCaptureSession_Tr2(s string, c string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QMediaCaptureSession_Tr2(s_Cstring, c_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QMediaCaptureSession_Tr3(s string, c string, n int) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QMediaCaptureSession_Tr3(s_Cstring, c_Cstring, (int)(n))
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QMediaCaptureSession) callVirtualBase_MetaObject() *qt.QMetaObject {
	return qt.UnsafeNewQMetaObject(unsafe.Pointer(QMediaCaptureSession_virtualbase_MetaObject(unsafe.Pointer(this.h))))
}

func (this *QMediaCaptureSession) OnMetaObject(slot func(super func() *qt.QMetaObject) *qt.QMetaObject) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QMediaCaptureSession_override_virtual_MetaObject(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_MetaObject
func miqt_exec_callback_QMediaCaptureSession_MetaObject(self QMediaCaptureSession, cb intptr_t) *QMetaObject {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func() *qt.QMetaObject) *qt.QMetaObject)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	virtualReturn := gofunc((&QMediaCaptureSession{h: self}).callVirtualBase_MetaObject)

	return (*QMetaObject)(virtualReturn.UnsafePointer())
}

func (this *QMediaCaptureSession) callVirtualBase_Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))

	return (unsafe.Pointer)(QMediaCaptureSession_virtualbase_Metacast(unsafe.Pointer(this.h), param1_Cstring))
}

func (this *QMediaCaptureSession) OnMetacast(slot func(super func(param1 string) unsafe.Pointer, param1 string) unsafe.Pointer) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QMediaCaptureSession_override_virtual_Metacast(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QMediaCaptureSession_Metacast
func miqt_exec_callback_QMediaCaptureSession_Metacast(self QMediaCaptureSession, cb intptr_t, param1 *const_char) unsafe.Pointer {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(param1 string) unsafe.Pointer, param1 string) unsafe.Pointer)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	param1_ret := param1
	slotval1 := GoString(param1_ret)

	virtualReturn := gofunc((&QMediaCaptureSession{h: self}).callVirtualBase_Metacast, slotval1)

	return virtualReturn
}

File: qt/multimedia/gen_qmediacapturesession.cpp
// +build ignore

#include <QAudioBufferInput>
#include <QAudioInput>
#include <QAudioOutput>
#include <QCamera>
#include <QImageCapture>
#include <QMediaCaptureSession>
#include <QMediaRecorder>
#include <QMetaObject>
#include <QObject>
#include <QScreenCapture>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QVideoFrameInput>
#include <QVideoSink>
#include <QWindowCapture>
#include <qmediacapturesession.h>
#include "gen_qmediacapturesession.h"
class MiqtVirtualQMediaCaptureSession : public virtual QMediaCaptureSession {
public:
MiqtVirtualQMediaCaptureSession(): QMediaCaptureSession() {};
MiqtVirtualQMediaCaptureSession(QObject* parent): QMediaCaptureSession(parent) {};

virtual ~MiqtVirtualQMediaCaptureSession() = default;
// cgo.Handle value for overwritten implementation
	intptr_t handle__MetaObject = 0;
// Subclass to allow providing a Go implementation
	virtual const QMetaObject* metaObject() const override {
if (handle__MetaObject == 0) {
return QMediaCaptureSession::metaObject();
}
QMetaObject* callback_return_value = miqt_exec_callback_QMediaCaptureSession_MetaObject(const_cast<MiqtVirtualQMediaCaptureSession*>(this), handle__MetaObject);

		return callback_return_value;
	}
// Wrapper to allow calling protected method
	QMetaObject* virtualbase_MetaObject() const {

		return (QMetaObject*) QMediaCaptureSession::metaObject();

	}
// cgo.Handle value for overwritten implementation
	intptr_t handle__Metacast = 0;
// Subclass to allow providing a Go implementation
	virtual void* qt_metacast(const char* param1) override {
if (handle__Metacast == 0) {
return QMediaCaptureSession::qt_metacast(param1);
}
const char* sigval1 = (const char*) param1;

		void* callback_return_value = miqt_exec_callback_QMediaCaptureSession_Metacast(this, handle__Metacast, sigval1);

		return callback_return_value;
	}
// Wrapper to allow calling protected method
	void* virtualbase_Metacast(const char* param1) {

		return QMediaCaptureSession::qt_metacast(param1);

	}
};
QMediaCaptureSession* QMediaCaptureSession_new() {
return new MiqtVirtualQMediaCaptureSession();
}
QMediaCaptureSession* QMediaCaptureSession_new2(QObject* parent) {
return new MiqtVirtualQMediaCaptureSession(parent);
}
void QMediaCaptureSession_virtbase(QMediaCaptureSession* src
, QObject** outptr_QObject
) {
*outptr_QObject = static_cast<QObject*>(src);
}
QMetaObject* QMediaCaptureSession_MetaObject(const QMediaCaptureSession* self) {
	return (QMetaObject*) self->metaObject();
}
void* QMediaCaptureSession_Metacast(QMediaCaptureSession* self, const char* param1) {
	return self->qt_metacast(param1);
}
struct miqt_string QMediaCaptureSession_Tr(const char* s) {
	QString _ret = QMediaCaptureSession::tr(s);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
QAudioInput* QMediaCaptureSession_AudioInput(const QMediaCaptureSession* self) {
	return self->audioInput();
}
void QMediaCaptureSession_SetAudioInput(QMediaCaptureSession* self, QAudioInput* input) {
	self->setAudioInput(input);
}
QAudioBufferInput* QMediaCaptureSession_AudioBufferInput(const QMediaCaptureSession* self) {
	return self->audioBufferInput();
}
void QMediaCaptureSession_SetAudioBufferInput(QMediaCaptureSession* self, QAudioBufferInput* input) {
	self->setAudioBufferInput(input);
}
QCamera* QMediaCaptureSession_Camera(const QMediaCaptureSession* self) {
	return self->camera();
}
void QMediaCaptureSession_SetCamera(QMediaCaptureSession* self, QCamera* camera) {
	self->setCamera(camera);
}
QImageCapture* QMediaCaptureSession_ImageCapture(QMediaCaptureSession* self) {
	return self->imageCapture();
}
void QMediaCaptureSession_SetImageCapture(QMediaCaptureSession* self, QImageCapture* imageCapture) {
	self->setImageCapture(imageCapture);
}
QScreenCapture* QMediaCaptureSession_ScreenCapture(QMediaCaptureSession* self) {
	return self->screenCapture();
}
void QMediaCaptureSession_SetScreenCapture(QMediaCaptureSession* self, QScreenCapture* screenCapture) {
	self->setScreenCapture(screenCapture);
}
QWindowCapture* QMediaCaptureSession_WindowCapture(QMediaCaptureSession* self) {
	return self->windowCapture();
}
void QMediaCaptureSession_SetWindowCapture(QMediaCaptureSession* self, QWindowCapture* windowCapture) {
	self->setWindowCapture(windowCapture);
}
QVideoFrameInput* QMediaCaptureSession_VideoFrameInput(const QMediaCaptureSession* self) {
	return self->videoFrameInput();
}
void QMediaCaptureSession_SetVideoFrameInput(QMediaCaptureSession* self, QVideoFrameInput* input) {
	self->setVideoFrameInput(input);
}
QMediaRecorder* QMediaCaptureSession_Recorder(QMediaCaptureSession* self) {
	return self->recorder();
}
void QMediaCaptureSession_SetRecorder(QMediaCaptureSession* self, QMediaRecorder* recorder) {
	self->setRecorder(recorder);
}
void QMediaCaptureSession_SetVideoOutput(QMediaCaptureSession* self, QObject* output) {
	self->setVideoOutput(output);
}
QObject* QMediaCaptureSession_VideoOutput(const QMediaCaptureSession* self) {
	return self->videoOutput();
}
void QMediaCaptureSession_SetVideoSink(QMediaCaptureSession* self, QVideoSink* sink) {
	self->setVideoSink(sink);
}
QVideoSink* QMediaCaptureSession_VideoSink(const QMediaCaptureSession* self) {
	return self->videoSink();
}
void QMediaCaptureSession_SetAudioOutput(QMediaCaptureSession* self, QAudioOutput* output) {
	self->setAudioOutput(output);
}
QAudioOutput* QMediaCaptureSession_AudioOutput(const QMediaCaptureSession* self) {
	return self->audioOutput();
}
void QMediaCaptureSession_AudioInputChanged(QMediaCaptureSession* self) {
	self->audioInputChanged();
}
void QMediaCaptureSession_connect_AudioInputChanged(QMediaCaptureSession* self, intptr_t slot) {
	MiqtVirtualQMediaCaptureSession::connect(self, static_cast<void (QMediaCaptureSession::*)()>(&QMediaCaptureSession::audioInputChanged), self, [=]() {
		miqt_exec_callback_QMediaCaptureSession_AudioInputChanged(slot);
	});
}
void QMediaCaptureSession_AudioBufferInputChanged(QMediaCaptureSession* self) {
	self->audioBufferInputChanged();
}
void QMediaCaptureSession_connect_AudioBufferInputChanged(QMediaCaptureSession* self, intptr_t slot) {
	MiqtVirtualQMediaCaptureSession::connect(self, static_cast<void (QMediaCaptureSession::*)()>(&QMediaCaptureSession::audioBufferInputChanged), self, [=]() {
		miqt_exec_callback_QMediaCaptureSession_AudioBufferInputChanged(slot);
	});
}
void QMediaCaptureSession_CameraChanged(QMediaCaptureSession* self) {
	self->cameraChanged();
}
void QMediaCaptureSession_connect_CameraChanged(QMediaCaptureSession* self, intptr_t slot) {
	MiqtVirtualQMediaCaptureSession::connect(self, static_cast<void (QMediaCaptureSession::*)()>(&QMediaCaptureSession::cameraChanged), self, [=]() {
		miqt_exec_callback_QMediaCaptureSession_CameraChanged(slot);
	});
}
void QMediaCaptureSession_ScreenCaptureChanged(QMediaCaptureSession* self) {
	self->screenCaptureChanged();
}
void QMediaCaptureSession_connect_ScreenCaptureChanged(QMediaCaptureSession* self, intptr_t slot) {
	MiqtVirtualQMediaCaptureSession::connect(self, static_cast<void (QMediaCaptureSession::*)()>(&QMediaCaptureSession::screenCaptureChanged), self, [=]() {
		miqt_exec_callback_QMediaCaptureSession_ScreenCaptureChanged(slot);
	});
}
void QMediaCaptureSession_WindowCaptureChanged(QMediaCaptureSession* self) {
	self->windowCaptureChanged();
}
void QMediaCaptureSession_connect_WindowCaptureChanged(QMediaCaptureSession* self, intptr_t slot) {
	MiqtVirtualQMediaCaptureSession::connect(self, static_cast<void (QMediaCaptureSession::*)()>(&QMediaCaptureSession::windowCaptureChanged), self, [=]() {
		miqt_exec_callback_QMediaCaptureSession_WindowCaptureChanged(slot);
	});
}
void QMediaCaptureSession_VideoFrameInputChanged(QMediaCaptureSession* self) {
	self->videoFrameInputChanged();
}
void QMediaCaptureSession_connect_VideoFrameInputChanged(QMediaCaptureSession* self, intptr_t slot) {
	MiqtVirtualQMediaCaptureSession::connect(self, static_cast<void (QMediaCaptureSession::*)()>(&QMediaCaptureSession::videoFrameInputChanged), self, [=]() {
		miqt_exec_callback_QMediaCaptureSession_VideoFrameInputChanged(slot);
	});
}
void QMediaCaptureSession_ImageCaptureChanged(QMediaCaptureSession* self) {
	self->imageCaptureChanged();
}
void QMediaCaptureSession_connect_ImageCaptureChanged(QMediaCaptureSession* self, intptr_t slot) {
	MiqtVirtualQMediaCaptureSession::connect(self, static_cast<void (QMediaCaptureSession::*)()>(&QMediaCaptureSession::imageCaptureChanged), self, [=]() {
		miqt_exec_callback_QMediaCaptureSession_ImageCaptureChanged(slot);
	});
}
void QMediaCaptureSession_RecorderChanged(QMediaCaptureSession* self) {
	self->recorderChanged();
}
void QMediaCaptureSession_connect_RecorderChanged(QMediaCaptureSession* self, intptr_t slot) {
	MiqtVirtualQMediaCaptureSession::connect(self, static_cast<void (QMediaCaptureSession::*)()>(&QMediaCaptureSession::recorderChanged), self, [=]() {
		miqt_exec_callback_QMediaCaptureSession_RecorderChanged(slot);
	});
}
void QMediaCaptureSession_VideoOutputChanged(QMediaCaptureSession* self) {
	self->videoOutputChanged();
}
void QMediaCaptureSession_connect_VideoOutputChanged(QMediaCaptureSession* self, intptr_t slot) {
	MiqtVirtualQMediaCaptureSession::connect(self, static_cast<void (QMediaCaptureSession::*)()>(&QMediaCaptureSession::videoOutputChanged), self, [=]() {
		miqt_exec_callback_QMediaCaptureSession_VideoOutputChanged(slot);
	});
}
void QMediaCaptureSession_AudioOutputChanged(QMediaCaptureSession* self) {
	self->audioOutputChanged();
}
void QMediaCaptureSession_connect_AudioOutputChanged(QMediaCaptureSession* self, intptr_t slot) {
	MiqtVirtualQMediaCaptureSession::connect(self, static_cast<void (QMediaCaptureSession::*)()>(&QMediaCaptureSession::audioOutputChanged), self, [=]() {
		miqt_exec_callback_QMediaCaptureSession_AudioOutputChanged(slot);
	});
}
struct miqt_string QMediaCaptureSession_Tr2(const char* s, const char* c) {
	QString _ret = QMediaCaptureSession::tr(s, c);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
struct miqt_string QMediaCaptureSession_Tr3(const char* s, const char* c, int n) {
	QString _ret = QMediaCaptureSession::tr(s, c, static_cast<int>(n));
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
void QMediaCaptureSession_override_virtual_MetaObject(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQMediaCaptureSession*>( (QMediaCaptureSession*)(self) )->handle__MetaObject = slot;
}
QMetaObject* QMediaCaptureSession_virtualbase_MetaObject(const void* self) {
	return ( (const MiqtVirtualQMediaCaptureSession*)(self) )->virtualbase_MetaObject();
}
void QMediaCaptureSession_override_virtual_Metacast(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQMediaCaptureSession*>( (QMediaCaptureSession*)(self) )->handle__Metacast = slot;
}
void* QMediaCaptureSession_virtualbase_Metacast(void* self, const char* param1) {
	return ( (MiqtVirtualQMediaCaptureSession*)(self) )->virtualbase_Metacast(param1);
}
void QMediaCaptureSession_Delete(QMediaCaptureSession* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<MiqtVirtualQMediaCaptureSession*>( self );
	} else {
		delete self;
	}
}
